### PR TITLE
Preliminary services 

### PR DESCRIPTION
**Pull request by lucasbotteri**
This PR adds the basic services that we'll need for basic functionalities, these services are not set in stone, and need a lot of work to be finished, there's a loot of room for improvement. But adding this basic modeling will enable us to unblock work and ease adding new stuff.

The following services were added, including model, hooks, types, services, relationships

- Character
- Campaign Template
- Game Engine
- Asset
- Event
- EventTemplate
- Roll
- ChatMessage

Update old models to add new relationships

Also enable plop template generation (they were commented and was not working) 
Improved templating

### CODE REVIEW

File: plopfile.js [hunk 1]
```diff
@@ -188,22 +188,22 @@ module.exports = function (plop) {
...
+
+      modifySequelizeImport,
+      modifySequelizeModelNames,
+      modifySequelizeModels,
+
+      modifyServiceConfigurationImport,
+      modifyServiceConfiguration
     ],
   });
 };
```
Comment (naueru) [inline plopfile.js]: New line please

File: src/models/campaignTemplates.model.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+    },
+    {
+      sequelize,
+      tableName: ModelName.CAMPAIGN_TEMPLATE,
+      modelName: ModelName.CAMPAIGN_TEMPLATE,
+      hooks: {
+        beforeCount(options: any) {
```
Comment (naueru) [inline src/models/campaignTemplates.model.ts:53]: There is not a more accurate type for this one?

File: src/services/assets/assets.class.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { AssetModel } from "../../models/assets.model";
+import { AssetDTO } from "./assets.dto";
+
+export class AssetService extends Service<AssetDTO> {
+  Model!: AssetModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/assets/assets.class.ts]: New line

File: src/services/assets/assets.dto.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export interface AssetDTO {
+  id: number;
+  url: string;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/assets/assets.dto.ts]: New Line

File: src/services/assets/assets.service.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.ASSETS }`, new AssetService(options, app));
+
+  const service = app.service(ServiceName.ASSETS);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/assets/assets.service.ts]: New line

File: src/services/campaignTemplates/campaignTemplates.class.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import { CampaignTemplateModel } from "../../models/campaignTemplates.model";
+import { CampaignTemplateDTO } from "./campaignTemplates.dto";
+
+export class CampaignTemplateService extends Service<CampaignTemplateDTO> {
+  Model!: CampaignTemplateModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/campaignTemplates/campaignTemplates.class.ts]: New line

File: src/services/campaignTemplates/campaignTemplates.dto.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export interface CampaignTemplateDTO {
+  id: number;
+  name: string;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/campaignTemplates/campaignTemplates.dto.ts]: New line

File: src/services/campaignTemplates/campaignTemplates.service.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.CAMPAIGN_TEMPLATES }`, new CampaignTemplateService(options, app));
+
+  const service = app.service(ServiceName.CAMPAIGN_TEMPLATES);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/campaignTemplates/campaignTemplates.service.ts]: New line

File: src/services/characters/characters.class.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+import { CharacterModel } from "../../models/characters.model";
+import { CharacterDTO } from "./characters.dto";
+
+export class CharacterService extends Service<CharacterDTO> {
+  Model!: CharacterModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/characters/characters.class.ts]: New line

File: src/services/characters/characters.dto.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+export interface CharacterDTO {
+  id: number;
+  name: string;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/characters/characters.dto.ts]: New line

File: src/services/characters/characters.service.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.CHARACTERS }`, new CharacterService(options, app));
+
+  const service = app.service(ServiceName.CHARACTERS);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/characters/characters.service.ts]: New line

File: src/services/chatMessages/chatMessages.class.ts [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+import { ChatMessageModel } from "../../models/chatMessages.model";
+import { ChatMessageDTO } from "./chatMessages.dto";
+
+export class ChatMessageService extends Service<ChatMessageDTO> {
+  Model!: ChatMessageModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/chatMessages/chatMessages.class.ts]: New line

File: src/services/chatMessages/chatMessages.dto.ts [hunk 13]
```diff
@@ -0,0 +1,7 @@
+export interface ChatMessageDTO {
+  id: number;
+  message: string;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/chatMessages/chatMessages.dto.ts]: New line

File: src/services/chatMessages/chatMessages.service.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.CHAT_MESSAGES }`, new ChatMessageService(options, app));
+
+  const service = app.service(ServiceName.CHAT_MESSAGES);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/chatMessages/chatMessages.service.ts]: New line

File: src/services/eventTemplates/eventTemplates.class.ts [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+import { EventTemplateModel } from "../../models/eventTemplates.model";
+import { EventTemplateDTO } from "./eventTemplates.dto";
+
+export class EventTemplateService extends Service<EventTemplateDTO> {
+  Model!: EventTemplateModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/eventTemplates/eventTemplates.class.ts]: New line

File: src/services/eventTemplates/eventTemplates.dto.ts [hunk 16]
```diff
@@ -0,0 +1,7 @@
+export interface EventTemplateDTO {
+  id: number;
+  command: string;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/eventTemplates/eventTemplates.dto.ts]: Liu nine

File: src/services/eventTemplates/eventTemplates.service.ts [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.EVENT_TEMPLATES }`, new EventTemplateService(options, app));
+
+  const service = app.service(ServiceName.EVENT_TEMPLATES);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/eventTemplates/eventTemplates.service.ts]: New line

File: src/services/events/events.class.ts [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+import { EventModel } from "../../models/events.model";
+import { EventDTO } from "./events.dto";
+
+export class EventService extends Service<EventDTO> {
+  Model!: EventModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/events/events.class.ts]: New line

File: src/services/events/events.dto.ts [hunk 19]
```diff
@@ -0,0 +1,7 @@
+export interface EventDTO {
+  id: number;
+  command: string;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/events/events.dto.ts]: New line

File: src/services/events/events.service.ts [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.EVENTS }`, new EventService(options, app));
+
+  const service = app.service(ServiceName.EVENTS);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/events/events.service.ts]: New line

File: src/services/gameEngines/gameEngines.class.ts [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+import { GameEngineModel } from "../../models/gameEngines.model";
+import { GameEngineDTO } from "./gameEngines.dto";
+
+export class GameEngineService extends Service<GameEngineDTO> {
+  Model!: GameEngineModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/gameEngines/gameEngines.class.ts]: New line

File: src/services/gameEngines/gameEngines.dto.ts [hunk 22]
```diff
@@ -0,0 +1,7 @@
+export interface GameEngineDTO {
+  id: number;
+  name: string;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/gameEngines/gameEngines.dto.ts]: New line

File: src/services/gameEngines/gameEngines.service.ts [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.GAME_ENGINES }`, new GameEngineService(options, app));
+
+  const service = app.service(ServiceName.GAME_ENGINES);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/gameEngines/gameEngines.service.ts]: New line

File: src/services/rolls/rolls.class.ts [hunk 24]
```diff
@@ -0,0 +1,12 @@
...
+import { RollModel } from "../../models/rolls.model";
+import { RollDTO } from "./rolls.dto";
+
+export class RollService extends Service<RollDTO> {
+  Model!: RollModel;
+
+  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+    super(options);
+  }
+}
```
Comment (naueru) [inline src/services/rolls/rolls.class.ts]: New line

File: src/services/rolls/rolls.dto.ts [hunk 25]
```diff
@@ -0,0 +1,7 @@
+export interface RollDTO {
+  id: number;
+  faces: number;
+
+  createdAt: Date;
+  updatedAt: Date;
+}
```
Comment (naueru) [inline src/services/rolls/rolls.dto.ts]: New line

File: src/services/rolls/rolls.service.ts [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+    Model: createModel(app),
+    paginate: app.get(SettingName.PAGINATE),
+  };
+
+  app.use(`/${ServiceName.ROLLS }`, new RollService(options, app));
+
+  const service = app.service(ServiceName.ROLLS);
+
+  service.hooks(hooks);
+}
```
Comment (naueru) [inline src/services/rolls/rolls.service.ts]: New line

File: templates/dto.hbs [hunk 27]
```diff
@@ -1,9 +1,9 @@
...
 {{#each properties}}
-  {{camelCase this.propertyName}}: {{lowerCase this.propertyType}}
+  {{camelCase this.propertyName}}: {{lowerCase this.propertyType}};
 {{/each}}
 
-  public createdAt: Date;
-  public updatedAt: Date;
+  createdAt: Date;
+  updatedAt: Date;
 }
```
Comment (naueru) [inline templates/dto.hbs]: Ups!, new line

**Review by naueru — COMMENTED** [review]
A bunch of new lines missin, its related to plop template?. Maybe you will need to adjust it.

File: src/models/campaignTemplates.model.ts (hunk 2 above)
Comment (lucasbotteri) [reply src/models/campaignTemplates.model.ts:53]: Well this is the file as generated by feathers-sequelize, that doesn't have a great typescript integration. I can try to fix this, but i should be done in another MR as is not that simple

**Review by naueru — APPROVED** [review]

**Pull request merged**